css_html_js_minify/html_minifier.py

```diff
@@ -20,9 +20,22 @@ def condense_html_whitespace(html):
     '<i><b><a> test </a></b></i><br>'
     """  # first space between tags, then empty new lines and in-between.
     log.debug("Removing unnecessary HTML White Spaces and Empty New Lines.")
-    is_ok = "<textarea" not in html.lower() and "<pre" not in html.lower()
-    html = re.sub(r'>\s+<', '> <', html) if is_ok else html
-    return re.sub(r'\s{2,}|[\r\n]', ' ', html) if is_ok else html.strip()
+    #split the html on the <pre> and <textarea> tags, mantaining the separators
+    split = re.split('(<pre>|</pre>|<textarea>|</textarea>)', html)
```

Comment:
yeah, i managed to get a more relaiable solution, but it still can't handle comments right. At the end i decided to use this better version i wrote for my website and to pay atention not to put <pre...> or <textarea ...> in comments.